pkg/olm/olm.go

```diff
@@ -70,39 +79,73 @@ func CmdLocalUninstall() *cobra.Command {
 	return cmd
 }
 
+// CmdCSV returns a CLI command
+func CmdCSV() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "csv",
+		Short: "CSV commands",
+	}
+	cmd.AddCommand(
+		CmdCSVYaml(),
+	)
+	return cmd
+}
+
+// CmdCSVYaml returns a CLI command
+func CmdCSVYaml() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "yaml",
+		Short: "Show bundled CSV",
+		Run:   RunCSVYaml,
+	}
+	return cmd
+}
+
+// RunHubInstall runs a CLI command
 func RunHubInstall(cmd *cobra.Command, args []string) {
 	hub := LoadHubConf()
 	for _, obj := range hub.Objects {
 		util.KubeCreateSkipExisting(obj)
 	}
 }
 
+// RunHubUninstall runs a CLI command
 func RunHubUninstall(cmd *cobra.Command, args []string) {
 	hub := LoadHubConf()
 	for _, obj := range hub.Objects {
 		util.KubeDelete(obj)
 	}
 }
 
+// RunHubStatus runs a CLI command
 func RunHubStatus(cmd *cobra.Command, args []string) {
 	hub := LoadHubConf()
 	for _, obj := range hub.Objects {
 		util.KubeCheck(obj)
 	}
 }
 
+// RunLocalInstall runs a CLI command
 func RunLocalInstall(cmd *cobra.Command, args []string) {
 	panic("TODO implement olm.RunLocalInstall()")
 }
 
+// RunLocalUninstall runs a CLI command
 func RunLocalUninstall(cmd *cobra.Command, args []string) {
 	panic("TODO implement olm.RunLocalUninstall()")
 }
 
+// RunCSVYaml runs a CLI command
+func RunCSVYaml(cmd *cobra.Command, args []string) {
+	fmt.Print(bundle.File_deploy_olm_catalog_package_noobaa_operator_v1_1_0_clusterserviceversion_yaml)
```

Comment:
@guymguym thanks for jumping on this. This is enough for ocs-operator to use to start integration work.

I do have a few concerns about how this will work long term though. 

1. The csv isn't being re-generated as part of the build system, so changes to rbac/deployment config aren't picked up automatically (at least that's the way it appears to me)

2. The csv version is hard coded here, which seems to indicate it will fall behind as new noobaa releases occur. 

3. This one isn't a huge deal, but I'll point it out still. PR #32 created a template where variables are defined to represent things like the csv version and container image. When integrating with ocs-operator, the template forces us to be aware of what variables are required to generate a component-level operator (like noobaa) csv.

So for example, if you added some sort of env var to the Deployment that needed to be dynamically set to a specific value, the ocs-operator would be alerted (by the addition of a new templated value) that there's a new integration piece for noobaa to take into account.

Reply:
Hey @davidvossel, thank you for getting into the details of it!

> 1. The csv isn't being re-generated as part of the build system, so changes to rbac/deployment config aren't picked up automatically (at least that's the way it appears to me)

You are right. At first we used to generate it using operator-sdk in the build, but I saw there were properties in the CSV that were not controllable using `operator-sdk olm-catalog gen-csv` and therefore we stopped auto generating it to avoid overriding those properties that we filled in manually. I also tried using the flags such as `--csv-config` but the docs of it said it support very few properties. I left this incremental CSV generation pending for a tool - either that operator-sdk will provide or we will add. How does OCS operator plan to generate the CSV?

> 2. The csv version is hard coded here, which seems to indicate it will fall behind as new noobaa releases occur.

Creating new version of a CSV is a manual decision as far as I understand it. There is nothing automated in operator-sdk that detects if you should create a new version, whether it is semver-major/minor/patch, or can it just update an existing CSV version. I think that pushing a new CSV to community-operators repo will only lag behind releases unless there is either a disciplined manual release process or an automated one. At this point in time we are releasing manually.

> 3. This one isn't a huge deal, but I'll point it out still. PR #32 created a template where variables are defined to represent things like the csv version and container image. When integrating with ocs-operator, the template forces us to be aware of what variables are required to generate a component-level operator (like noobaa) csv.

If needed we can provide this with a CLI flag that will replace those fields with a go template indicator. But in any way I guess anyone can do that, it is a trivial replace like you did in the script.

Perhaps again I can pull for an advice from @dmesser on how to tackle this...